src/Strategy/DiscoveryStrategy.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Http\Discovery\Strategy;
+
+use Http\Discovery\Exception\StrategyUnavailableException;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+interface DiscoveryStrategy
+{
+    /**
+     * Find a resource of a specific type.
+     *
+     * @param string $type
+     *
+     * @return array The return value is always an array with zero or more elements. Each
+     *               element is an array with two keys ['class' => string, 'condition' => mixed].
+     *
+     * @throws StrategyUnavailableException if we cannot use this strategy.
+     */
+    public static function find($type);
```

Comment:
Maybe not use "find" here because it is too similar to to the `ClassDiscovery`. To avoid confusion this maybe should be renamed to "getBinding" or "getClasses".